circle.yml

```diff
@@ -1,31 +1,15 @@
 machine:
-  pre:
-    # Install sbt 0.13.16
-    - sudo apt-get install openjdk-8-jdk
-    - wget -q https://dl.bintray.com/sbt/debian/sbt-0.13.16.deb
-    - sudo dpkg -i sbt-0.13.16.deb
-  cache_directories:
-    - "~/.ivy2"
-    - "~/.sbt"
+  services:
+    - docker
 
 dependencies:
-  pre:
-    - sudo apt-get update
-    - sudo apt-get install libffi-dev libssl-dev python-dev
-    - sudo apt-get install python
-    - python --version
-    - curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
-    - sudo python get-pip.py
-    - sudo pip install --upgrade pip
-    - sudo pip install --upgrade setuptools
-    - sudo pip install pyopenssl ndg-httpsclient pyasn1
-    - sudo pip install moto --ignore-installed six
-    - sudo pip install flask
-    - moto_server s3 -p 8001 &
+  override:
+    - docker info
+    - docker build --rm=false -t circleci/moztelemetry:$CIRCLE_SHA1 .
 
 test:
   override:
-    - 'true'
+    - docker run -t -i -p 8001:8001 circleci/moztelemetry:$CIRCLE_SHA1 ./bin/run_tests
```

Comment:
Will the `sbt publish` work correctly? I'll tag hwoo for an r on this as well.

Reply:
You can test sbt publish locally.  If that works it should work

Reply:
I think we'll need to publish inside the container? I'm not very familiar with what `sbt publish` does. If so, I think the command we'll need in this circle file is: `docker run -ti circleci/moztelemetry:$CIRCLE_SHA1 sbt publish`.

I ran that locally and got an error:
`[error] (*:publish) java.lang.UnsupportedOperationException: URL repository only support HTTP PUT at the moment`

Reply:
the publish should be executed outside the container.